src/Microsoft.DotNet.ProjectModel/Graph/LockFile.cs

```diff
@@ -87,5 +84,18 @@ public bool IsValidForProject(Project project, out string message)
             message = null;
             return true;
         }
+
+        public void MergeWith(LockFile fragmentLockFile)
+        {
+            if (Version != fragmentLockFile.Version)
+            {
+                throw new ArgumentException($"Cannot merge master lock file {LockFilePath} with fragment lock file {fragmentLockFile.LockFilePath} due to missmatching versions");
+            }
+
+            Targets.MergeWith(fragmentLockFile.Targets, l => l.TargetFramework.Framework + l.RuntimeIdentifier);
```

Comment:
When main lock has this: "targets": {    ".NETFramework,Version=v4.6": {

and fragment has   "targets": {    ".NETFramework,Version=v4.5.2": {

after it merges it would create 2 targets sections? and how it would work after that? My target app has target net46, my reference app has target 4.5.2, they are compatible but different. Should main lock file add 4.5.2 under 4.6 ?